File: bc-scm-web/src/main/webapp/app/app/view/mailhistory/MailHistoryController.js
/**
 * Copyright @ 2015 nanjing wanghua Information Technology Co.,Ltd.
 * All right reserved.
 *
 * @author: chengxuemin
 * date: 2015-12-09
 */

/**
 * 邮件历史记录控制器
 */
Ext.define('app.view.mailhistory.MailHistoryController', {
    extend: 'app.view.base.BaseController',
    alias: 'controller.mailhistory',
    init: function(view){
        var viewmodel = view.getViewModel();
        this.getInternationalPro({config:'MailHistoryConfig'},viewmodel);
    },
    /**
     * 查询邮件
     * @param v
     */
    query:function(v){

        // TODO 防止频繁点击刷新，禁用按钮
        var form = v.up('mailhistoryqueryform');

        if(true != form.isValid()){
            return;
        }
        var grid = form.nextSibling("mailhistorygrid");
        var store = grid.getStore();
        var params = form.getForm().getValues();
        Ext.apply(store.proxy.extraParams, params);
        store.load();
    },

    /**
     * 添加按钮
     * @param btn
     */
    add:function(btn){
        Ext.create('app.view.mailhistory.MailHistoryInfoWin').show();
    },

    /**
     * 查看
     * @param btn
     */
    check: function(btn){
        var grid = btn.up('mailhistorygrid');
        // 获取选中的列
        var record = grid.getSelectionModel().getSelection();
        if(this.oneSelectVali(record)){
            var mailhistoryWin = Ext.create('app.view.mailhistory.MailHistoryInfoWin',{title:this.getData('checkEmailContent'),stateId:'check'}).show();
            mailhistoryWin.down('form').loadRecord(record[0]);
        }
    }
});

File: bc-scm-web/src/main/java/com/beautifulchina/controller/mailhistory/MailHistoryController.java
/**
 * Copyright @ 2015 nanjing wanghua Information Technology Co.,Ltd.
 * All right reserved.
 *
 * @author: chengxuemin
 * date: 2015-12-11
 */
package com.beautifulchina.controller.mailhistory;

import com.beautifulchina.base.BaseController;
import com.beautifulchina.common.Page;
import com.beautifulchina.mailhistory.vo.MailHistoryVO;
import com.beautifulchina.main.bo.MailHistoryListBO;
import com.beautifulchina.service.mailhistory.MailHistoryService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;

import javax.validation.Valid;
/**
 * 邮件历史记录控制器
 */
@Controller
@RequestMapping(value = "/mailhistory")
//后端的入口，传入参数返回后端的结果
public class MailHistoryController extends BaseController {

    private static Logger logger = LoggerFactory.getLogger(MailHistoryController.class);

    @Autowired
    private MailHistoryService mailhistoryService;

    /**
     * 查询邮件列表
     * @param mailhistoryVO
     * @return
     */
    @RequestMapping(value = "query")
    @ResponseBody
    public Page<MailHistoryListBO> queryMailHistorys(@Valid MailHistoryVO mailhistoryVO){
        Page<MailHistoryListBO> page = mailhistoryService.getAllMailHistoryList(mailhistoryVO);
        return page;
    }
}
